view/frontend/web/js/view/payment/method-renderer/mondu.js

```diff
@@ -1,134 +1,158 @@
-                if (event) {
-                    event.preventDefault();
-                }
+        openCheckout: function (token) {
+            $(
+              '<div id="mondu-checkout-widget" style="position: fixed; top: 0;right: 0;left: 0;bottom: 0; z-index: 99999999;"/>'
+            ).appendTo("body");
+            window.monduCheckout.render({
+                token,
+                onCancel: () => {
+                    $("#mondu-checkout-widget").remove();
+                    self.isPlaceOrderActionAllowed(true);
+                    $("body").trigger("processStop");
+                },
+                onSuccess: () => {
+                    self.getPlaceOrderDeferredObject()
+                      .fail(function () {
+                          self.isPlaceOrderActionAllowed(true);
+                          $("body").trigger("processStop");
+                      })
+                      .done(function () {
+                          self.afterPlaceOrder();
+                          if (self.redirectAfterPlaceOrder) {
+                              redirectOnSuccessAction.execute();
+                          }
+                      });
+                    $("#mondu-checkout-widget").remove();
+                    $("body").trigger("processStop");
+                },
+                onClose: () => {},
+            });
+        },
 
-                $('body').trigger('processStart');
+        placeOrder: function (data, event) {
+            if (!additionalValidators.validate()) return;
+            if (event) {
+                event.preventDefault();
+            }
+            if (self.isBillingSameAsShippng) {
+                quote.billingAddress(quote.shippingAddress());
+            }
+            $("body").trigger("processStart");
 
+            var initCheckout = function () {
                 $.ajax({
                     url: self.getMonduCheckoutTokenUrl(),
-                    method: 'get',
+                    method: "get",
                     data: {
-                        email: self.getCustomerEmail()
-                    }
-                }).always(function(res) {
-                    if(res && res.token && !res.error) {
+                        email: self.getCustomerEmail(),
+                    },
+                }).always(function (res) {
+                    if (res && res.token && !res.error) {
                         self.openCheckout(res.token);
                         return;
                     } else {
                         self.messageContainer.addErrorMessage({
-                            message: res.message
+                            message: res.message,
                         });
                     }
 
-                    $('body').trigger('processStop');
+                    $("body").trigger("processStop");
                 });
-            }
-        });
-    }
-);
+            };
+            SetPaymentInformationAction(
```

Comment:
Update checkout information manually before opening our widget ( in case of some checkout flows like in onestepcheckout )